llvmlite/ir/types.py

```diff
@@ -257,6 +257,17 @@ def _as_float(value):
     return struct.unpack('f', struct.pack('f', value))[0]
 
 
+def _as_half(value):
+    """
+    Truncate to half-precision float.
+    """
+    try:
+        return struct.unpack('e', struct.pack('e', value))[0]
+    except struct.error:
+        # 'e' only added in Python 3.6+
+        return _as_float(value)
```

Comment:
Might this be a source of confusion, a user asks for a half but when formatted it appears to not have been correctly truncated?

Reply:
I could use numpy to round it ([the logic is complicated!](https://bugs.python.org/issue11734)), but I don't think llvmlite depends on that at the minute. Is that OK? I'd rather use a library to do the rounding - especially as python 2.x is decreasing in popularity.

Reply:
@sklam do you have a view on this. Perhaps just documenting that it's different or issuing a warning is the easiest thing to do?

Reply:
If not rounded properly, at `parse_assembly()`, it will raise `RuntimeError` with:

```python
RuntimeError: LLVM IR parsing error
<string>:5:27: error: floating point constant invalid for type
@"myhalf" = constant half 0x3f87006680000000
```

I think we should just document it.  Telling user to consider doing something like `float(np.float16(1.12312321e-2))` to make sure the value can be exactly represented as half.

Reply:
@sklam @stuartarchibald I've pushed a change to the documentation that covers this